python/pyarrow/_compute.pyx

```diff
@@ -1142,6 +1142,36 @@ class PadOptions(_PadOptions):
         self._set_options(width, padding, lean_left_on_odd_padding)
 
 
+cdef class _ZeroFillOptions(FunctionOptions):
+    def _set_options(self, width, padding):
+        self.wrapped.reset(new CZeroFillOptions(width, tobytes(padding)))
+
+
+class ZeroFillOptions(_ZeroFillOptions):
+    """
+    Options for utf8_zero_fill.
+
+    Parameters
+    ----------
+    width : int
+        Desired string length.
+    padding : str, default "0"
+        Padding character. Should be one byte or Unicode codepoint.
```

Comment:
There is no Ascii version of this function, so we can simplify the description just like in C++.
```suggestion
        Padding character. Should be one Unicode codepoint.
```